Dictionary.java

```diff
@@ -0,0 +1,49 @@
+import java.util.LinkedList;
+
+public class Dictionary {
+	private LinkedList<String> words;
+	private LinkedList<String> prefixes;
+
+	public Dictionary(LinkedList<String> words) {
+		this.words = words;
+		this.prefixes = new LinkedList<String>();

```

Comment:
I would suggest extracting the computing part outside of the constructor. It is generally bad practice to have any work done in the constructor. I suggest you arrange it in the following way: the constructor only initializes the required fields, then we provide a method that sets "words" list and computes "prefixes" list.